src/server/my-server.js

```diff
@@ -32,8 +32,14 @@ class Server {
             log(`Server started on ${this.host}:${this.port}`);
         }
         catch (err) {
-            error(`Server failed to start`);
-
+            switch (err.code) {
```

Comment:
It's nice that you try to display different error messages. But outputting the original error can be even more helpful.

Reply:
After switch statement I'm throw the error which should be caught by global error-handler. It's no more than additional useful info,